src/app/AuthenticationControllerContainer.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { GrafanaInfo, KialiAppState, LoginStatus } from '../store/Store';
+import * as API from '../services/Api';
+import { HelpDropdownActions } from '../actions/HelpDropdownActions';
+import { JaegerActions } from '../actions/JaegerActions';
+import { MessageCenterActions } from '../actions/MessageCenterActions';
+import { MessageType } from '../types/MessageCenter';
+import { KialiDispatch } from '../types/Redux';
+import { ServerStatus } from '../types/ServerStatus';
+import { GrafanaActions } from '../actions/GrafanaActions';
+import InitializingScreen from './InitializingScreen';
+import { isKioskMode } from '../utils/SearchParamUtils';
+import * as MessageCenter from '../utils/MessageCenter';
+import { setServerConfig } from '../config/serverConfig';
+
+interface AuthenticationFlowProps {
+  authenticated: boolean;
+  protectedAreaComponent: React.ReactNode;
+  publicAreaComponent: React.ReactNode;
+  setGrafanaInfo: (grafanaInfo: GrafanaInfo) => void;
+  setServerStatus: (serverStatus: ServerStatus) => void;
+}
+
+interface AuthenticationFlowState {
+  stage: 'login' | 'post-login' | 'logged-in';
```

Comment:
This would have better type-safety and reusability if it were an **enum**.

Reply:
out of curiosity: why? I've always thought union types were actually a nice alternative to enums (it can be declared explicitly as a type as well)

Reply:
I would like to keep it as is for now, since state is kind of private. WDYT? We could convert it to a type if the stage moves to the properties.

Reply:
If it's local I guess it's fine.